Fix shutdown deadlock introduced in 2.7.0-M2 (#8815)
playframework/playframework: documentation/manual/working/commonGuide/shutdown/Shutdown.md

Java
: @[shutdown-task](code/shutdown/ResourceAllocatingJavaClass.java)

## Shutdown triggers

A Play process is usually terminated via a `SIGTERM` signal. When the signal is received by the Play process a JVM shutdown hook is run causing the server to stop via invoking Coordinated Shutdown.

> Other possible triggers differ from `SIGTERM` slightly. While `SIGTERM` is handled in an outside-in fashion you may trigger a shutdown from your code (or a library may detect a cause to trigger the shutdown). One such example is running your Play process as part of an Akka Cluster, or adding an endpoint on your API that would allow an admin or an orchestrator to trigger a programmatic shutdown. In these scenarios the shutdown is inside out: all the phases of the Coordinated Shutdown list are run in the appropriate order but the Actor System will terminate before the JVM shutdown hook runs.

Reviewer: ```suggestion
Other possible triggers differ from `SIGTERM` slightly. While `SIGTERM` is handled in an outside-in fashion, you may trigger a shutdown from your code (or a library may detect a cause to trigger the shutdown). For example when running your Play process as part of an Akka Cluster or adding an endpoint on your API that would allow an admin or an orchestrator to trigger a programmatic shutdown. In these scenarios the shutdown is inside out: all the phases of the Coordinated Shutdown list are run in the appropriate order, but the Actor System will terminate before the JVM shutdown hook runs.
```

> When developing your Play application you should consider all the termination triggers and what steps and in which order they will run.

Reviewer: ```suggestion
When developing your Play application, you should consider all the termination triggers and what steps and in which order they will run.
```
